### DISCUSSION TITLE
How to set overlay text/image opacity with watermark?

### DISCUSSION THREAD

**Discussion by zhaohuxing** [libvips/libvips#2526] [Q&A]
Hi @jcupitt, I don't find args with `vips_composite2` and `vips_text` to set opacity. how to write ?😊

https://github.com/libvips/libvips/discussions/2522

**Comment by jcupitt** [comment]
Hi again,

You can just scale the alpha down, for example:

```
text *= [1, 1, 1, 0.3]
```

So RGB stays the same, but A is multiplied by 0.3.

**Reply by zhaohuxing** [reply]
`text` type is VipsImage, how to set `text *= [1, 1, 1, 0.3]`?

**Reply by jcupitt** [reply]
In Lua it would be:

```
text = text * {1, 1, 1, 0.3}
```

**Reply by zhaohuxing** [reply]
call [`vips_image_new_from_image`](https://www.libvips.org/API/current/VipsImage.html#vips-image-new-from-image) ? can you give me demo with c or rust? 😊

**Reply by jcupitt** [reply]
Ah, sorry, I thought you were using Lua.

In C, perhaps:

```C
VipsImage *in = vips_text (...);
VIpsImage *x;

double scale[4] = {1, 1, 1, 0.3};
double offset[4] = {0, 0, 0, 0};
if (vips_linear(in, &x, scale, offset, NULL))
    error_handler ();
g_object_unref (in);
in = x;
```

**Reply by zhaohuxing** [reply]
Thanks ❤️
